tests/api/utils/activitypub.js

```diff
@@ -0,0 +1,79 @@
+const _ = require('builders/utils')
+const express = require('express')
+const { createUser, createUsername, createUserOnFediverse } = require('../fixtures/users')
+const { randomActivity } = require('./activities')
+const makeUrl = require('controllers/activitypub/lib/make_url')
+
+const endpoint = '/api/activitypub'
+
+const createReceiver = async (customData = {}) => {
+  const username = createUsername()
+  const userAttributes = _.extend({
+    username,
+    fediversable: true
+  }, customData)
+  return createUser(userAttributes)
+}
+
+const query = ({ action, username }) => `${endpoint}?action=${action}&name=${username}`
+
+const startServerWithEmitterAndReceiver = async (params = {}) => {
+  let { emitterUser } = params
+  if (!emitterUser) emitterUser = await createUserOnFediverse()
+  const { origin, query } = await startServerWithEmitterUser({ emitterUser })
+  const keyUrl = makeUrl({ origin, params: query })
+  const { username } = await createReceiver()
+  const privateKey = emitterUser.privateKey
+  const receiverUrl = makeUrl({ params: { action: 'actor', name: username } })
+  return { origin, keyUrl, privateKey, receiverUrl, receiverUsername: username }
+}
+
+const startServerWithEmitterUser = async ({ emitterUser }) => {
+  const { origin } = await startActivityPubServer({ user: emitterUser })
+  return {
+    origin,
+    query: { action: 'actor', name: emitterUser.username }
+  }
+}
+
+const startActivityPubServer = ({ user, endpoints = [] }) => new Promise(resolve => {
+  const port = 1024 + Math.trunc(Math.random() * 10000)
```

Comment:
wouldn't a single server be enough? do we need a different server for each of those tests?

Reply:
done in 465a36c